spec/mailers/candidate_mailer_spec.rb

```diff
@@ -242,4 +253,56 @@
       'course start' => 'September 2021',
     )
   end
+
+  describe '.new_interview' do
+    let(:email) { mailer.new_interview(application_choice_with_interview, interview) }
+
+    before do
+      @provider_name = application_choice_with_interview.offered_course.provider.name
```

Comment:
This can be moved to a let within the interview block, which should also enable to remove the duplicate entries specified below.

Reply:
I tried that before but for some reason I'm getting
```
'provider_name' is not available on an example group (e.g. a `describe` or `context` block). 
It is only available from within individual examples (e.g. `it` blocks) or from constructs that 
run in the scope of an example (e.g. `before`, `let`, etc).
```
when I defined it with `let` underneath `let(:application_choice_with_interview)`

Reply:
I didn't notice that the variable was being referenced from inside the shared examples when I reviewed! let declarations are not available from within shared examples due to potential issues that may be caused, such as records are created outside example transactions.

However I think we can still avoid requiring the variable by just defining the provider and not doing any evaluations within the examples below.

```
 let(:provider)  { create(:provider, name: 'Brighthurst Technical College') }
 let(:interview) do
    create(:interview,
           date_and_time: Time.zone.local(2021, 1, 15, 9, 30),
           location: 'Hogwarts Castle',
           additional_details: 'Bring your magic wand for the spells test',
           provider: provider)
 end
```

This will also require updating the interview factory (and also removing the invalid condition that I added, as there will always be an application choice!!) 

```
 factory :interview do
    application_choice

    date_and_time { 7.business_days.from_now }
    location { [Faker::Address.full_address, 'Link to video conference'].sample }
    additional_details { [nil, 'Use staff entrance', 'Ask for John at the reception'].sample }

    after(:build) do |interview|
      interview.application_choice.status = 'interviewing'
      interview.provider ||= interview.application_choice.offered_course.provider
    end
  end

```